### PR TITLE
Fix  extra ack being sent after a connection is established

### LINKED ISSUE #2829: [BUG] Extra ACK is sent at the start of a connection (due to an uninitialized variable)
After PR #2745 once a connection is established both sides (sender and receiver) without any reason send the first full ACK with the initial sequence number.

### To Reproduce

Start a Wireshark capture and run:

```shell
srt-xtransmit generate "srt://:4200" --enable-metrics -v --sendrate 5Mbps --duration 3s

srt-xtransmit receive "srt://127.0.0.1:4200" --enable-metrics -v
```

In the capture right after the handshake you will see those two useless ACKs.
image

#### SRT Version Affected : v1.5.3.

### PR DESCRIPTION
**Pull request by yomnes0**
Add an initialization of m_bBufferWasFull to false when starting a connection. This will prevent an extra ack from being sent right after the handshake.

Fixes #2829.

### CODE REVIEW

File: srtcore/core.h [hunk 1]
```diff
@@ -980,6 +980,10 @@ class CUDT
     CallbackHolder<srt_listen_callback_fn> m_cbAcceptHook;
     CallbackHolder<srt_connect_callback_fn> m_cbConnectHook;
 
+    void resetBufferWasFull() { 
+        m_bBufferWasFull = FALSE;
```
Comment (ethouris) [inline srtcore/core.h:984]: `false`

Comment (ethouris) [reply srtcore/core.h:984]: Another thing - why do you need this intermediate method here? You can just as well set this directly in `CUDT::startConnect`...

Comment (yomnes0) [reply srtcore/core.h:984]: True, I thought that going through a method was better than direct access, I'll update this

**Comment by ethouris** [comment]
Ah, forgot one thing: if you set this to true in `startConnect`, shouldn't it be also the same way initialized when accepted out of the listener?

**Comment by yomnes0** [comment]
That's true, I was focusing on the issue on the transmitter side and forgot about the receiver

File: docs/API/API-socket-options.md [hunk 2]
```diff
@@ -474,7 +474,7 @@ The default flow control window size is 25600 packets. It is approximately:
 - 300 Mbits of payload with an SRT payload size of 1456 bytes.
 
 The minimum number of packets in flight should be (assuming max payload size):
-`FCmin = bps / 8 × RTTsec / (MSS - 44)`,
+`FCmin = bps / 8 × latency_sec / (MSS - 44)`,
```
Comment (maxsharabayko) [inline docs/API/API-socket-options.md:477]: This change does not belong to this PR.
RTTsec is correct here.

**Comment by maxsharabayko** [comment]
@yomnes0 Could you please nae PRs describing what is changed/fixed. This lands in the commit message. "Fix for bug #2829" tells nothing, except for the need to go to GitHub to find the issue and figure out what's fixed.

File: srtcore/core.h (hunk 1 above)
Comment (maxsharabayko) [reply srtcore/core.h]: The initialization is needed not just for SRT caller, but in general.

File: docs/API/API-socket-options.md (hunk 2 above)
Comment (yomnes0) [reply docs/API/API-socket-options.md:477]: This was supposed to go in another PR, I must have forgotten to unmark it before commiting

Comment (maxsharabayko) [reply docs/API/API-socket-options.md:477]: Although it should probably be `RTTsec + latency_sec` if we include unacknowledged packets in the RCV buffer.
Still, in a separate PR.

**Review by maxsharabayko — CHANGES_REQUESTED** [review]
The `startConnect` function is used by a caller (and probably by a rendezvous), but not by a listener. The variable stays uninitialized for a listener.
@yomnes0 please use the common place for initialization where all other `CUDT` member variables are initialized.

File: srtcore/core.cpp [hunk 3]
```diff
@@ -3555,7 +3556,7 @@ void srt::CUDT::startConnect(const sockaddr_any& serv_addr, int32_t forced_isn)
     m_iISN = m_ConnReq.m_iISN = forced_isn;
 
     setInitialSndSeq(m_iISN);
-
+
```
Comment (maxsharabayko) [inline srtcore/core.cpp:3559]:
```suggestion

```

**Review by maxsharabayko — APPROVED** [review]

**Pull request merged**